src/modules/patternReplace/patternReplace.module.ts

```diff
@@ -32,10 +52,34 @@ export const patternReplace: BotModule = {
       const hasModification = message.content !== modifiedContent;
       if (!hasModification) return;
 
-      const newMessage = [`<@${message.author.id}>`, modifiedContent].join('\n');
+      const newMessageContent = [`<@${message.author.id}>`, modifiedContent].join('\n');
+
+      const newMessage = await message.channel.send(newMessageContent);
+
+      await newMessage.react('❌');
+      await message.react('❌');
+
+      const awaitReactionsOptions: AwaitReactionsOptions = {
+        filter: (reaction, user) => reactionCollectorFilter(reaction, user, message.author.id),
+        max: 1,
+        time: 60000,
+        errors: ['time'],
+      };
+
+      message
+        .awaitReactions(awaitReactionsOptions)
+        .then(() => deleteMessage(message))
+        .catch(console.error);
+
+      newMessage
+        .awaitReactions(awaitReactionsOptions)
+        .then(() => deleteMessage(newMessage))
+        .catch(console.error);
 
-      await message.channel.send(newMessage);
-      await message.delete();
+      void wait(60000).then(async () => {
+        await message.reactions.cache.get('❌')?.remove();
+        await newMessage.reactions.cache.get('❌')?.remove();
+      });
```

Comment:
what happens if the bot restarts during the timeout?, I think that the function who handle the deletion can and need to be decorrelated from the creation

Reply:
yes, it's what i put in the first message of the PR, I linked a documentation that talks about it. I need to solve this problem